Tabl info.inc.php moved to table class #12567

flash1452 opened the pull request:
Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

Signed-off-by: flash1452 vishpandey2014@gmail.com

flash1452 committed:
shift the tbl_info.inc.php code to Table class
shift the tbl_info.inc.php code to Table class
shift the tbl_info.inc.php code to Table class

shift the tbl_info.inc.php code to Table class

commit to resolve lint error

substituting tbl_info_inc.php in tbl_operations.php

Added tests for new functions in Table class

delete tbl_info.in.php and shift code to Table class

delete tbl_info.in.php and shift code to Table class

undo changes in database password

trying to improve code coverage. Signed-off-by: flash1452 vishpandey2014@gmail.com

Remove extra $ fro. Signed-off-by: flash1452 vishpandey2014@gmail.com

issue - call getTable function just once.

Signed-off-by: Vishal Pandey <vishpandey2014@gmail.com>

issue - call getTable function just once.                                                        Signed-Off-By: flash1452 <vishpandey2014@gmail.com>

Removing additional newlines and adjusting whitespaces.
Signed-off-by: Vishal Pandey <vishpandey2014@gmail.com>

Removing parameter  and using internal functions
Signed-off-by: Vishal Pandey <vishpandey2014@gmail.com>

changes in num of rows function
Signed-off-by: Vishal Pandey vishpandey2014@gmail.com

 occurences removed and test cases modifed to accomodate comparison with false.

modifying  usage and refactoring code

modifying  usage and refactoring code

flash1452 commented:
@nijel kindly review my PR and suggest changes.

flash1452 commented:
@nijel Also if you can help me with this code coverage problem.

nijel commented on libraries/Response.php:
```diff
@@ -1,5 +1,5 @@
 <?php
-/* vim: set expandtab sw=4 ts=4 sts=4: */
+/* vim: set expandtab sw=4 ts=4 sts=4:  */
```
I guess this change is not intentional.

nijel commented on libraries/controllers/table/TableChartController.php:
```diff
@@ -113,7 +114,60 @@ public function indexAction()
+            //$tbl_object = new Table($GLOBALS['db'], $GLOBALS['table']);
+            $showtable = $GLOBALS['dbi']->getTable(
+                                        $GLOBALS['db'],
+                                        $GLOBALS['table']
```
You should really do the getTable call just once and then use the resulting class in subsequent calls.

flash1452 replied:
@nijel Yeah, you are damn right there is no point in calling getTable function so many times.

nijel reviewed (changes requested):
I assume phpunit.phar file should not be there, please remove it.

Also the repeated getTable calls should be removed - it should be called just once and then use resulting Table object.

flash1452 commented:
@nijel I have made some of the changes that you suggested but code coverage still remains an issue.

flash1452 commented:
@nijel Should I consider a new PR?? Because Code coverage is not passing.

nijel commented on libraries/Table.php:
```diff
@@ -321,6 +322,104 @@ public function getStatusInfo(
+    *
+    * @return   string               Return storage engine info if it is set for *                                the selected table else return blank.
+    */
+    public function getTableStorageEngine($showtable){
```
It really should use `$this->getStatusInfo('ENGINE', false, true)` instead of getting the data as parameter.

nijel commented on libraries/Table.php:
```diff
@@ -321,6 +322,104 @@ public function getStatusInfo(
+        if($showtable){
+
+            return isset($showtable['Comment']) ? $showtable['Comment'] : '';
+        }
```
Same here, you should rely on class data and not get it from elsewhere as this can lead to errors.

Use `$this->getStatusInfo('COMMENT', false, true)` instead.

nijel commented on libraries/Table.php:
```diff
@@ -321,6 +322,104 @@ public function getStatusInfo(
+    public function getTableCollation($showtable){
+        if($showtable){
+
+            return empty($showtable['Collation']) ? '' : $showtable['Collation'];
```
Same here, you should rely on class data and not get it from elsewhere as this can lead to errors.

Use `$this->getStatusInfo('COLLATION', false, true)` instead.

nijel commented on libraries/Table.php:
```diff
@@ -321,6 +322,104 @@ public function getStatusInfo(
+    * @param Current table properties.
+    * @return Return no of rows info if it is not null for the selected table or return 0.
+    */
+    public function getTableNumRowInfo($showtable){
```
Again, do not rely on passed parameters.

nijel commented on libraries/controllers/table/TableChartController.php:
```diff
@@ -113,7 +114,34 @@ public function indexAction()
+            $showtable = $table_class_object->getStatusInfo(
+                null,
+                (isset($reread_info) && $reread_info ? true : false)
+            );
```
You really should not need this, the Table class should handle this internally.

nijel commented on libraries/controllers/table/TableIndexesController.php:
```diff
@@ -63,8 +63,35 @@ public function indexAction()
+        $showtable = $table_class_object->getStatusInfo(
+            null,
+            (isset($reread_info) && $reread_info ? true : false)
+        );
```
...and same here.

nijel commented:
Yes most of the code is not covered by tests, but that really does not matter for now as it was previously not covered as well

flash1452 commented:
@nijel Made the changes you suggested.

flash1452 commented:
@nijel Kindly review the updated PR. The code coverage has improved drastically.

nijel commented:
Thanks! I will get to it next week.

nijel commented on libraries/Menu.php:
```diff
@@ -231,8 +231,19 @@ private function _getBreadcrumbs()
-                include './libraries/tbl_info.inc.php';
-
+                // include './libraries/tbl_info.inc.php';
+                $reread_info = true;
```
This does not seem to be used anywhere now. You should instead forcing of reading the information again here, for example using:

```php
$table_class_object->getStatusInfo(null, true);
```

nijel replied:
What about this variable? You set it in several places but it really doesn't seem to be used. In case the rereading is really necessary here, you should actually reread the information as mentioned in above comment.

nijel commented on libraries/Table.php:
```diff
@@ -321,6 +322,110 @@ public function getStatusInfo(
+    */
+    public function getShowComment() {
+        $table_comment = $this->getStatusInfo('COMMENT', false, true);
+        if ($table_comment) {
```
You probably want to check this as `=== false` instead to avoid surprises with PHP automatic typecasting. Without this you would be ignoring comment consisting of `0`, try yourself:

```php

if ('0') {
    echo "HERE\n";
}
```

Same applies to other methods as well.

flash1452 commented:
@nijel I have made some of the suggested changes in this PR and also squashed all the commits.

flash1452 commented:
@nijel I forgot, Congratulations on getting selected for this years GSoC.

flash1452 commented:
I have made the changes related to rereading. But the code coverage seems to have taken a hit.

flash1452 commented:
@nijel Please review the PR.

nijel committed:
Merge remote-tracking branch 'origin/pull/12931' into tmp

nijel committed:
Simplify rereading of information

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Remove unused variables

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Remove commented out code

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Fix comments indentation

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting storage engine

- remove not needed word Table from method name
- fix code to handle errors
- convert existing users of getStatusInfo('ENGINE') to this method

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting table comment

- remove not needed word Show from method name
- fix code to handle errors

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting table collation

- remove not needed word Table from method name
- fix code to handle errors

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting table rows

- simplify method name
- fix code handling error path

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting row format

- remove not needed word Table from method name
- fix code to handle errors
- convert existing users of getStatusInfo('ROW_FORMAT') to this method

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Simplify getting table create options

- use saner method name (getCreateOptions)
- fix error case

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Remove documentation for non existing param

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel committed:
Document return types

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel commented:
I did quick review and this is really far from being complete.

Following files do still use tbl_info.inc.php which you've removed:

* libraries/controllers/table/TableIndexesController.php
* libraries/controllers/table/TableRelationController.php
* libraries/controllers/table/TableSearchController.php
* libraries/controllers/table/TableStructureController.php
* libraries/mult_submits.inc.php
* tbl_addfield.php
* tbl_export.php
* tbl_find_replace.php
* tbl_import.php
* tbl_operations.php
* tbl_select.php
* tbl_sql.php
* tbl_structure.php
* tbl_zoom_select.php

There are some other things to improve in the code which I've fixed (I thought I could merge the code with such changes prior to realizing above):

* ba0f9785c3 Document return types
* 4c4f851d7a Remove documentation for non existing param
* bbef205ca4 Simplify getting table create options
* 5dbd529e69 Simplify getting row format
* f92cc062d4 Simplify getting table rows
* 8f121757c6 Simplify getting table collation
* 2344b79f70 Simplify getting table comment
* ff1427020d Simplify getting storage engine
* 1ddd344fb5 Fix comments indentation
* a90f2f2a08 Remove commented out code
* a7e6c07188 Remove unused variables
* 6899e8f6da Simplify rereading of information

And most important question: Have you actually tested the code you've written? Even table structure does not work with your changes...

flash1452 commented:
@nijel Thanks for you review. I will have to accept I have been careless while testing my code but yeah, I will  soon make the required changes.

flash1452 commented:
@nijel Please correct me if I am wrong but libraries/controllers/table/TableIndexesController.php,
libraries/controllers/table/TableRelationController.php, libraries/controllers/table/TableSearchController.php doesn't seem to be using any of the variables being set in tbl_info.inc.php.

flash1452 commented:
@nijel There are many such files where the lib_info.inc.php is included but the variables being set in lib_info.inc.php is not used. One such example is TableRelationController.php where these variables are passed as parameters to the constructor. What should be the way to handle these situations?

nijel commented:
Yes, it can be that the variables are not used there, I didn't look at the code, I've just noticed that it's still included. You need to go through these files and check whether they need some of the variables...

flash1452 committed:
Removed additional ocurrences of tbl_info.inc.php

flash1452 commented:
@nijel I have updated the code with your mentioned changes. I have tested my code this time but still if you can test it and review it.

flash1452 committed:
Removed additional ocurrences of tbl_info.inc.php

nijel committed:
Merge branch 'tabl_info.inc.php-moved-to-Table-class' into table_info

flash1452 committed:
Merge pull request #1 from nijel/table_info

Fixes to table info

flash1452 commented:
@nijel There is a link broken in docs, that's why the TRAVIS build is failing.

 (line   41) broken    https://anonscm.debian.org/cgit/collab-maint/phpmyadmin.git/tree/debian/README.Debian - HTTPSConnectionPool(host='anonscm.debian.org', port=443): Max retries exceeded with url: /cgit/collab-maint/phpmyadmin.git/tree/debian/README.Debian (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x2b233231d8d0>: Failed to establish a new connection: [Errno 113] No route to host',)).

Should we remove this url from the doc or replace it.

nijel commented:
It was temporary failure, I've just restarted the job.

flash1452 commented:
@nijel Still the code coverage seems to drop. But, I guess this should not make much of difference?

flash1452 commented:
@nijel What can be done to improve the code coverage? Any suggestions.

nijel committed:
Merge remote-tracking branch 'origin/pull/12931' into tmp

nijel committed:
Fix method names

Signed-off-by: Michal Čihař <michal@cihar.com>

nijel commented:
Have you tested the code? I got several errors, fixed them in f88971e34045eff61db397a0f33cfd48f2d69d43, you can merge it at https://github.com/flash1452/phpmyadmin/pull/2

As for coverage, it's probably okay, but you should really test it manually as well as the automated tests still don't cover everything...

flash1452 committed:
Merge pull request #2 from nijel/fix-12931

Another update

flash1452 commented:
@nijel I am getting a Error 500 whenever I am trying to test my phpmyadmin branch. 
The apache error logs show - 
Class 'PhpMyAdmin\\MoTranslator\\Loader' not found in /usr/share/phpmyadmin/libraries/common.inc.php on line 90.

But the weird thing is my old version of the same branch works perfectly and in that version in common.inc.php instead of 'PhpMyAdmin\\MoTranslator\\Loader' only 'MoTranslator\Loader' has been used.

flash1452 committed:
Revert "Another update"

flash1452 committed:
Merge pull request #3 from flash1452/revert-2-fix-12931

Revert "Another update"

nijel commented:
You need to run composer, see https://docs.phpmyadmin.net/en/latest/setup.html#installing-from-git

flash1452 committed:
Revert "Merge pull request #3 from flash1452/revert-2-fix-12931"

This reverts commit 89ec36f52454d67a13e5b238f3bef71a59879d3a, reversing
changes made to f9bb66917da6d523309b6dc270011bd11badec37.

flash1452 commented:
@nijel I have tried and tested the current version  of code with my changes. So, if you can have a look at it and give it a run.

nijel commented:
Thanks, I've manually merged and rebased your changes.
